core/arch/arm/plat-vexpress/main.c

```diff
@@ -32,55 +32,21 @@
 
 #include <drivers/gic.h>
 #include <drivers/pl011.h>
-#include <sm/sm.h>
-#include <sm/tee_mon.h>
-
-#include <util.h>
 
 #include <arm.h>
-#include <kernel/thread.h>
-#include <kernel/panic.h>
+#include <kernel/common_boot.h>
 #include <kernel/pm_stubs.h>
 #include <trace.h>
 #include <kernel/misc.h>
 #include <kernel/tee_time.h>
 #include <mm/tee_pager.h>
-#include <mm/core_mmu.h>
-#include <mm/tee_mmu.h>
-#include <mm/tee_mm.h>
-#include <utee_defines.h>
-#include <tee/tee_cryp_provider.h>
 #include <tee/entry.h>
 #include <tee/arch_svc.h>
 #include <console.h>
-#include <malloc.h>
-#include <assert.h>
-
-#define PADDR_INVALID		0xffffffff
-
-extern uint8_t __text_init_start[];
-extern uint8_t __data_start[];
-extern uint8_t __data_end[];
-extern uint8_t __bss_start[];
-extern uint8_t __bss_end[];
-extern uint8_t __init_start[];
-extern uint8_t __init_size[];
-extern uint8_t __heap1_start[];
-extern uint8_t __heap1_end[];
-extern uint8_t __heap2_start[];
-extern uint8_t __heap2_end[];
-extern uint8_t __pageable_part_start[];
-extern uint8_t __pageable_part_end[];
-extern uint8_t __pageable_start[];
-extern uint8_t __pageable_end[];
 
 static void main_fiq(void);
-#if defined(CFG_WITH_ARM_TRUSTED_FW)
-/* Implemented in assembly, referenced in this file only */
-uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-#endif
 
-static const struct thread_handlers handlers = {
+const struct thread_handlers handlers = {
```

Comment:
Can we have an accessor to `handlers` so that it is not used directly in other files? And also have a better for it (`handlers_common_boot`  for ex.) ?